lib/src/zap/rdma/zap_rdma.c

```diff
@@ -2278,18 +2278,19 @@ static int send_credit_update(struct z_rdma_ep *rep)
 		     "rem_rq_credits %d.\n",
 		     __func__, rep->sq_credits, rep->lcl_rq_credits,
 		     rep->rem_rq_credits);
+		pthread_mutex_unlock(&rep->credit_lock);
+		pthread_mutex_lock(&rep->ep.lock);
 		__rdma_context_free(ctxt);
 		__rdma_buffer_free(rbuf);
-		rc = ENOSPC;
 		pthread_mutex_unlock(&rep->ep.lock);
+		pthread_mutex_lock(&rep->credit_lock);
+		rc = ENOSPC;
```

Comment:
If we're going to do this, do we need the credit lock at all?

Reply:
@tom95858 we probably won't. But at this point I'm afraid to pull credit_lock out and break everything 😅 .